### ISSUE TITLE
Tests are not finishing with NUnit 3.13.1 and dotnet 5

### ISSUE THREAD

**Issue by xadvfh** [reportportal/agent-net-nunit#70]
Tests are not finishing with NUnit 3.13.1 and dotnet 5.

Report portal logs contain the following errors:

```
ReportPortalListener Error: 0 : 10:09:47.7933722 : 1-testhost : ReportPortal exception was thrown.
System.ArgumentException: Cannot parse empty value.
   at ReportPortal.Shared.Execution.Metadata.MetaAttribute.Parse(String value)
   at ReportPortal.NUnitExtension.ReportPortalListener.FinishTest(String report) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Test.cs:line 88
ReportPortalListener Error: 0 : 10:09:47.8002702 : 1-testhost : ReportPortal exception was thrown.
System.InsufficientExecutionStackException: Some of child test item(s) are not scheduled to finish yet.
   at ReportPortal.Shared.Reporter.TestReporter.Finish(FinishTestItemRequest request)
   at ReportPortal.NUnitExtension.ReportPortalListener.<FinishSuite>b__45_0(String __id, FinishTestItemRequest __finishSuiteRequest, String __report, String __parentstacktrace) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 191
   at ReportPortal.NUnitExtension.ReportPortalListener.FinishSuite(String report) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 222
ReportPortalListener Error: 0 : 10:09:47.8026000 : 1-testhost : ReportPortal exception was thrown.
System.InsufficientExecutionStackException: Some of child test item(s) are not scheduled to finish yet.
   at ReportPortal.Shared.Reporter.TestReporter.Finish(FinishTestItemRequest request)
   at ReportPortal.NUnitExtension.ReportPortalListener.<FinishSuite>b__45_0(String __id, FinishTestItemRequest __finishSuiteRequest, String __report, String __parentstacktrace) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 191
   at ReportPortal.NUnitExtension.ReportPortalListener.FinishSuite(String report) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 222
ReportPortalListener Error: 0 : 10:09:47.8037242 : 1-testhost : ReportPortal exception was thrown.
System.InsufficientExecutionStackException: Some of child test item(s) are not scheduled to finish yet.
   at ReportPortal.Shared.Reporter.TestReporter.Finish(FinishTestItemRequest request)
   at ReportPortal.NUnitExtension.ReportPortalListener.<FinishSuite>b__45_0(String __id, FinishTestItemRequest __finishSuiteRequest, String __report, String __parentstacktrace) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 191
   at ReportPortal.NUnitExtension.ReportPortalListener.FinishSuite(String report) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 222
ReportPortalListener Error: 0 : 10:09:47.8048896 : 1-testhost : ReportPortal exception was thrown.
System.InsufficientExecutionStackException: Some of child test item(s) are not scheduled to finish yet.
   at ReportPortal.Shared.Reporter.TestReporter.Finish(FinishTestItemRequest request)
   at ReportPortal.NUnitExtension.ReportPortalListener.<FinishSuite>b__45_0(String __id, FinishTestItemRequest __finishSuiteRequest, String __report, String __parentstacktrace) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 191
   at ReportPortal.NUnitExtension.ReportPortalListener.FinishSuite(String report) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 222
ReportPortalListener Error: 0 : 10:09:47.8060875 : 1-testhost : ReportPortal exception was thrown.
System.InsufficientExecutionStackException: Some of child test item(s) are not scheduled to finish yet.
   at ReportPortal.Shared.Reporter.TestReporter.Finish(FinishTestItemRequest request)
   at ReportPortal.NUnitExtension.ReportPortalListener.<FinishSuite>b__45_0(String __id, FinishTestItemRequest __finishSuiteRequest, String __report, String __parentstacktrace) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 191
   at ReportPortal.NUnitExtension.ReportPortalListener.FinishSuite(String report) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Suite.cs:line 222
ReportPortalListener Error: 0 : 10:09:47.8107531 : 1-testhost : ReportPortal exception was thrown.
System.InsufficientExecutionStackException: Some of child test item(s) are not scheduled to finish yet.
   at ReportPortal.Shared.Reporter.LaunchReporter.Finish(FinishLaunchRequest request)
   at ReportPortal.NUnitExtension.ReportPortalListener.FinishRun(String report) in C:\projects\agent-net-nunit\src\ReportPortal.NUnitExtension\ReportPortalListener.Launch.cs:line 110
```

**Comment by nvborisenko** [comment]
@xadvfh do you think the issue is related to version of nunit or .net framework? I cannot reproduce it on my side using the same set of versions.

This issue looks like to be simple resolvable, but please share simple test project to reproduce it.

**Comment by xadvfh** [comment]
@nvborisenko i'm pretty sure it is related to dotnet 5.  We have other projects that are still on dotnetcore3.1 using the latest nunit and they work fine.  

Also, I downgraded to my projects to dotnetcore3.1 and everything started working again.

**Comment by nvborisenko** [comment]
@xadvfh based on agent code I can imagine that nunit engine provides empty test category. I was able to reproduce this issue in following way

```csharp
[Caterory("")]
public void Test() {}
```

Looks like you create empty category for test. Please confirm whether this is your case.

Definitely this issue should be fixed on agent side, I would like to understand if the fix will help to resolve your issue. Please get deep diver what exactly causes the error.

**Comment by xadvfh** [comment]
I don't have any empty categories but i do have a custom attribute that extends category attribute like this:

```
    [AttributeUsage(AttributeTargets.All, AllowMultiple = false)]
    public class OldNameAttribute : CategoryAttribute {
        public OldNameAttribute(string oldTestName) : base(oldTestName) { }
    }
```

also have some tests that are using multiple `Category` atributes.

**Comment by nvborisenko** [comment]
Yeah, this is the same. Please try to find the place where you put empty `oldTestName` string value. Hmm, but you mentioned it's related to .net 5. NUnit engine produces empty category somehow, please highlight how.

If it's possible to share nunit's result xml file, it will help to identify problematic test. Or, you can set `ReportPortal_TraceLevel` environment variable to `Verbose` (don't forget to restart VS or any other child processes to take an effect) and provide log files.

**Comment by xadvfh** [comment]
For some reason I can't get this to reproduce at the moment.  I will close for now and reopen when/if I can.  I'll then post the verbose logging you recommended.

**Comment by nvborisenko** [comment]
@xadvfh  I am targeting the fix to v4.1.0
Please let me know if this issue is still reproducible and noisy for you.
